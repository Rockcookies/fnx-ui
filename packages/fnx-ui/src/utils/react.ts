import { Children, ReactElement } from 'react';
import { isFragment } from 'react-is';

export const toElementArray = (
	children: React.ReactNode,
	option?: { keepEmpty?: boolean },
): ReactElement[] => {
	const keepEmpty = option && option.keepEmpty;
	let ret: React.ReactElement[] = [];

	Children.forEach(children, (child: any) => {
		if ((child === undefined || child === null) && !keepEmpty) {
			return;
		}

		if (Array.isArray(child)) {
			ret = ret.concat(toElementArray(child));
		} else if (isFragment(child) && child.props) {
			ret = ret.concat(toElementArray(child.props.children, option));
		} else {
			ret.push(child);
		}
	});

	return ret;
};
