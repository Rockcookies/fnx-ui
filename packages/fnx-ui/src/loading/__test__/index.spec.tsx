import React from 'react';
import { render } from '@testing-library/react';
import Loading from '../index';

describe('<Loading/>', () => {
	it('should set size correctly', () => {
		const { container } = render(<Loading size={20}>Loading...</Loading>);

		expect(container).toMatchSnapshot();
	});

	it('should set textSize correctly', () => {
		const { container } = render(
			<Loading textSize={20}>Loading...</Loading>,
		);

		expect(container).toMatchSnapshot();
	});

	it('should set color correctly', () => {
		const { container } = render(
			<Loading color="#1989fa">Loading...</Loading>,
		);

		expect(container).toMatchSnapshot();
	});

	it('should set textColor correctly', () => {
		const { container } = render(
			<Loading textColor="#1989fa">Loading...</Loading>,
		);

		expect(container).toMatchSnapshot();
	});
});
