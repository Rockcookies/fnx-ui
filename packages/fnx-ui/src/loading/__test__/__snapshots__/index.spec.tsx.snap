// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> should set color correctly 1`] = `
<div>
  <span
    class="fnx-loading"
    style="color: rgb(25, 137, 250);"
  >
    <span
      class="fnx-icon"
    >
      <svg
        aria-hidden="true"
        class="fnx-icon__spinner"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="25 25 50 50"
        width="1em"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="20"
        />
      </svg>
    </span>
    <span
      class="fnx-loading__text"
      style="color: rgb(25, 137, 250);"
    >
      Loading...
    </span>
  </span>
</div>
`;

exports[`<Loading/> should set size correctly 1`] = `
<div>
  <span
    class="fnx-loading"
  >
    <span
      class="fnx-icon"
      style="font-size: 20px;"
    >
      <svg
        aria-hidden="true"
        class="fnx-icon__spinner"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="25 25 50 50"
        width="1em"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="20"
        />
      </svg>
    </span>
    <span
      class="fnx-loading__text"
    >
      Loading...
    </span>
  </span>
</div>
`;

exports[`<Loading/> should set textColor correctly 1`] = `
<div>
  <span
    class="fnx-loading"
  >
    <span
      class="fnx-icon"
    >
      <svg
        aria-hidden="true"
        class="fnx-icon__spinner"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="25 25 50 50"
        width="1em"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="20"
        />
      </svg>
    </span>
    <span
      class="fnx-loading__text"
      style="color: rgb(25, 137, 250);"
    >
      Loading...
    </span>
  </span>
</div>
`;

exports[`<Loading/> should set textSize correctly 1`] = `
<div>
  <span
    class="fnx-loading"
  >
    <span
      class="fnx-icon"
    >
      <svg
        aria-hidden="true"
        class="fnx-icon__spinner"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="25 25 50 50"
        width="1em"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="20"
        />
      </svg>
    </span>
    <span
      class="fnx-loading__text"
      style="font-size: 20px;"
    >
      Loading...
    </span>
  </span>
</div>
`;
